sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -1821,6 +1821,47 @@ -(void) testIncrementalPullWithInsert
     XCTAssertEqual(client.syncContext.pendingOperationsCount, 1);
 }
 
+// Tests incremental pull with interleaved local operations is Id case insensitive
+-(void)testIncrementalPullIsIdCaseInsensitive
+{
+    // Response of a pull operation
+    NSString *firstPullData = @"[{\"id\": \"one\", \"text\":\"first item\", \"__updatedAt\":\"1999-12-03T15:44:29.0Z\"},{\"id\": \"two\", \"text\":\"second item\", \"__updatedAt\":\"1999-12-03T15:44:28.0Z\"}]";
+
+    // Setup filter to get the pull response
+    MSMultiRequestTestFilter *filter = [MSMultiRequestTestFilter testFilterWithStatusCodes:@[@200] data:@[firstPullData] appendEmptyRequest:YES];
+
+    // Initialization
+    MSClient *filteredClient = [client clientWithFilter:filter];
+    MSSyncTable *todoTable = [filteredClient syncTableWithName:@"TodoItem"];
+    MSQuery *query = [[MSQuery alloc] initWithSyncTable:todoTable];
+
+    // Perform an insert operation in the middle of a pull operation
+    ((MSTestFilter *)filter.testFilters[0]).onInspectRequest = ^(NSURLRequest *request) {
+
+        // Insert an item with Id differing from that of the pulled item only in case
+        NSDictionary *item = @{ @"id": @"ONE", @"name":@"fifth item" };
+        MSTestWaiter *insertWaiter = [MSTestWaiter new];
+    
+        [todoTable insert:item completion:^(NSDictionary *item, NSError *error) {
```

Comment:
There should be 2 tests here I believe:
1 - Pull -> Get id of "ABC" -> Verify its upsert into "abc" works fine.  This is the test I expected from the name of the method.  If you want to say this case is covered by the tests directly in MSCoreDataStore and not needed, I can accept that.

2 - A test of the merge when a pending operation is queued during the pull, which is what we this appears to be doing.  I don't think the inc pull matters in this flow.  This would just be testPullKeepsPendingChangesWithDifferentCasedId or some such.  This tests that the record retrieved from the server is discarded when there is already a pending op in the queue and not written back to disk and the pending op is merged correctly back in.